### ISSUE TITLE
Moving a project on disk breaks IntelliSense, etc due to caching of targets

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn-project-system#1538]
1. Clone https://github.com/Ponant/ResxBugVS2017RC
2. Open project
3. Notice that Dependencies node has errors and get Intellisense errors 

I debugged this and this is caused by the caching of the NuGet props, which results in the wrong paths being used for $(ProjectAssetsFile), for example, from a design-time build:
```
17>Target "ResolvePackageDependenciesForBuild" skipped, due to false condition; ('$(ResolvePackageDependenciesForBuild)' == 'true' and Exists('$(ProjectAssetsFile)')) was evaluated as ('true' == 'true' and Exists('U:\Calculus\Visual Studio 2017\Projects\ResBugWebAppCore11\ResBugWebAppCore11\obj\project.assets.json')).
```

This is despite this project being on my D:\. NuGet restore will re-generate the props/targets to point to the correct paths but Visual Studio will not pick up the new values until you restart it.

**Comment by emgarten** [comment]
The project.assets.json, targets, and props files should not be checked into source control.

NuGet could add only relative paths, but this would still break when trying to resolve packages from the global packages folder whose path is listed in the project.assets.json file.

For packages.config this worked because the packages folder was next to the solution. But since the global folder could be anywhere relative to where the code lives it doesn't really work to use relative paths.

**Comment by davkean** [comment]
This will occur regardless of whether it is checked into source control, ie just moving a project to another location.

**Comment by davkean** [comment]
The Packages root  uses %USERPROFILE% but the global packages folder hardcodes a path, why is that? https://github.com/Ponant/ResxBugVS2017RC/blob/master/ResBugWebAppCore11/ResBugWebAppCore11/obj/ResBugWebAppCore11.csproj.nuget.g.props.

**Comment by emgarten** [comment]
> The Packages root uses %USERPROFILE% but the global packages folder hardcodes a path, why is that? 

The props file contains a copy of the package folder values from project.assets.json

**Comment by davkean** [comment]
That doesn't answer my question, why is the environment variable used in one variable and not in another?

**Comment by emgarten** [comment]
MSBuild has support resolving environment variables, JSON doesn't. The feature to add macros to the props/targets file where you see UserProfile used came as a community PR to NuGet to help in some scenarios like the one you are seeing.

**Comment by srivatsn** [comment]
So this is just https://github.com/dotnet/project-system/issues/268 right?

**Comment by davkean** [comment]
No, it's more than #268. We should be able to populate the language service from these assets files without a restore having to finish.

**Comment by nkolev92** [comment]
Maybe I'm missing something, but why wouldn't the below work? 
```xml
    <ProjectAssetsFile Condition=" '$(ProjectAssetsFile)' == '' ">$(MSBuildThisFileDirectory)project.assets.json</ProjectAssetsFile>
```

**Comment by jjmew** [comment]
@drewnoakes  As you mention this could be related to the issue when you rename a project and the Dependencies Node doesn't react to it.
